File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/PlayerController.java
package lt.ltrp;

import lt.ltrp.dao.PlayerDao;
import lt.ltrp.object.LtrpPlayer;
import net.gtaun.shoebill.data.Location;
import net.gtaun.shoebill.object.Destroyable;

import java.util.Collection;

/**
 * @author Bebras
 *         2016.04.08.
 */
public interface PlayerController extends Destroyable {

    static final int MINUTES_FOR_PAYDAY = 20;
    public static final Location GYM_LOCATION = new Location();

    class Instance
    {
        static PlayerController instance = null;
    }

    static PlayerController get()
    {
        return Instance.instance;
    }

    Collection<LtrpPlayer> getPlayers();
    PlayerDao getPlayerDao();
    String getUsernameByUUID(int uuid);


}

File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/player/PlayerController.kt
package lt.ltrp.player

import lt.ltrp.`object`.LtrpPlayer
import lt.ltrp.`object`.PlayerData
import net.gtaun.shoebill.`object`.Destroyable
import net.gtaun.shoebill.data.Location

/**
 * @author Bebras
 *         2016.04.08.
 */
abstract class PlayerController protected constructor(): Destroyable {

    init {
        instance = this
    }

    companion object {

        lateinit var instance: PlayerController

        fun get(): PlayerController {
            return instance
        }

        val MINUTES_FOR_PAYDAY = 20
        val GYM_LOCATION = Location()
    }

    abstract fun getPlayers(): Collection<LtrpPlayer>
    abstract fun getUsernameByUUID(uuid: Int): String?
    abstract fun getData(uuid: Int): PlayerData?
    abstract fun update(player: LtrpPlayer)


}
